winpr/libwinpr/thread/thread.c

```diff
@@ -911,11 +917,19 @@ HANDLE _GetCurrentThread(VOID)
 
 DWORD GetCurrentThreadId(VOID)
 {
+#if defined(__FreeBSD__)
+	int tid = pthread_getthreadid_np();
+	return tid;
```

Comment:
please use `WINPR_CXX_COMPAT_CAST` to cast to `DWORD` (aka unsigned int) to avoid sign warnings

Reply:
Done -- did the same for the Linux equivalent.